test/fakeintake/server/http.go

```diff
@@ -22,3 +25,53 @@ func writeHttpResponse(w http.ResponseWriter, response httpResponse) {
 		w.Write(response.body)
 	}
 }
+
+type errorResponse struct {
+	Errors []string `json:"errors"`
+}
+
+// Same struct as in datadog-agent/comp/core/flare/helpers/send_flare.go
+type flareResponse struct {
+	CaseID int    `json:"case_id,omitempty"`
+	Error  string `json:"error,omitempty"`
+}
+
+func buildErrorResponse(responseError error) httpResponse {
+	statusCode := http.StatusAccepted
+
+	resp := errorResponse{}
+	if responseError != nil {
+		statusCode = http.StatusBadRequest
+		resp.Errors = []string{responseError.Error()}
+	}
+
+	return buildResponse(resp, statusCode, "application/json")
+}
+
+func buildSuccessResponse(urlPath string) httpResponse {
+	var body interface{}
+	if urlPath == "/support/flare" {
+		body = flareResponse{}
+	} else {
+		body = errorResponse{}
+	}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
`http.go` should not know about routes, the caller should keep the knowledge and this function should take a body interface instead

Reply:
Done in cb7559bcd868c8abdf089273d7f5e3b6252d4cd6